demo-client/src/main/java/com/illia/client/service/query/processor/unit/SortOperationProcessorUnit.java

```diff
@@ -0,0 +1,120 @@
+package com.illia.client.service.query.processor.unit;
+
+
+import com.illia.client.model.IMDbMovieEntity;
+import com.illia.client.model.holder.IMDbMovieHolder;
+import com.illia.client.model.request.entity.QueryEntity;
+import com.illia.client.model.request.entity.SortQueryEntity;
+import com.illia.client.model.request.registry.AttributeRegistry;
+import com.illia.client.model.request.registry.OrderRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+
+import java.text.SimpleDateFormat;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@Slf4j
+public class SortOperationProcessorUnit implements OperationProcessor {
+    @Autowired
+    IMDbMovieHolder holder;
+    /**
+     * Sorts input entities list by 'attribute'
+     * in 'order' order and
+     * takes 'limit' elements
+     */
+
+    @Override
+    public List<IMDbMovieEntity> proceed(List<IMDbMovieEntity> records, QueryEntity queryEntity){
+        var attribute = ((SortQueryEntity) queryEntity).getAttribute();
```

Comment:
downcast the query into a new variable at the start of the method to not do this 3 times